File: codegen/oas-ehr/javascript-closure-angular/API/Client/AbstractEntry.js
goog.provide('API.Client.AbstractEntry');

/**
 * @record
 */
API.Client.AbstractEntry = function() {}

/**
 * @type {!API.Client.CodePhrase}
 * @export
 */
API.Client.AbstractEntry.prototype.language;

/**
 * @type {!API.Client.CodePhrase}
 * @export
 */
API.Client.AbstractEntry.prototype.encoding;

/**
 * @type {!Array<!API.Client.Participation>}
 * @export
 */
API.Client.AbstractEntry.prototype.otherParticipations;

/**
 * @type {!API.Client.ObjectRef}
 * @export
 */
API.Client.AbstractEntry.prototype.workflowId;

/**
 * @type {!API.Client.PartyProxy}
 * @export
 */
API.Client.AbstractEntry.prototype.subject;

/**
 * @type {!API.Client.PartyProxy}
 * @export
 */
API.Client.AbstractEntry.prototype.provider;

/**
 * @type {!string}
 * @export
 */
API.Client.AbstractEntry.prototype.type;

File: codegen/swagger-ehr/dart/lib/model/abstract_entry.dart
part of swagger.api;

class AbstractEntry {
  
  CodePhrase language = null;

  CodePhrase encoding = null;

  List<Participation> otherParticipations = [];

  ObjectRef workflowId = null;

  PartyProxy subject = null;

  PartyProxy provider = null;

  AbstractEntry();

  @override
  String toString() {
    return 'AbstractEntry[language=$language, encoding=$encoding, otherParticipations=$otherParticipations, workflowId=$workflowId, subject=$subject, provider=$provider, ]';
  }

  AbstractEntry.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    language = new CodePhrase.fromJson(json['language']);
    encoding = new CodePhrase.fromJson(json['encoding']);
    otherParticipations = Participation.listFromJson(json['other_participations']);
    workflowId = new ObjectRef.fromJson(json['workflow_id']);
    subject = new PartyProxy.fromJson(json['subject']);
    provider = new PartyProxy.fromJson(json['provider']);
  }

  Map<String, dynamic> toJson() {
    return {
      'language': language,
      'encoding': encoding,
      'other_participations': otherParticipations,
      'workflow_id': workflowId,
      'subject': subject,
      'provider': provider
     };
  }

  static List<AbstractEntry> listFromJson(List<dynamic> json) {
    return json == null ? new List<AbstractEntry>() : json.map((value) => new AbstractEntry.fromJson(value)).toList();
  }

  static Map<String, AbstractEntry> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, AbstractEntry>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new AbstractEntry.fromJson(value));
    }
    return map;
  }
}
